src/persist-client/src/impl/machine.rs

```diff
@@ -200,29 +200,48 @@ where
         }
     }
 
+    pub async fn listen(&self, as_of: &Antichain<T>) -> Result<Self, Since<T>> {
```

Comment:
This is super nit, but: it feels a bit weird to have these methods, because `ReadHandle` doesn't really have to call them, they're just an additional layer of verification/assertion. We could maybe put that in a comment here or maybe call this `verify_listen` or sth.